pkg/archive/archive_test.go

```diff
@@ -53,55 +55,51 @@ func defaultCopyWithTar(src, dst string) error {
 }
 
 func TestIsArchivePathDir(t *testing.T) {
-	cmd := exec.Command("sh", "-c", "mkdir -p /tmp/archivedir")
-	output, err := cmd.CombinedOutput()
-	if err != nil {
-		t.Fatalf("Fail to create an archive file for test : %s.", output)
-	}
-	if IsArchivePath(tmp + "archivedir") {
+	if IsArchivePath(t.TempDir()) {
 		t.Fatalf("Incorrectly recognised directory as an archive")
 	}
 }
 
 func TestIsArchivePathInvalidFile(t *testing.T) {
-	cmd := exec.Command("sh", "-c", "dd if=/dev/zero bs=1024 count=1 of=/tmp/archive && gzip --stdout /tmp/archive > /tmp/archive.gz")
+	tmpDir := t.TempDir()
+	cmd := exec.Command("sh", "-c", fmt.Sprintf("dd if=/dev/zero bs=1024 count=1 of=%[1]s/archive && gzip --stdout %[1]s/archive > %[1]s/archive.gz", toUnix(t, tmpDir)))
```

Comment:
No clue how these paths in `sh` translate to paths on the host; looks like the `TempDir` that's created is not available within `sh`;


```
=== FAIL: github.com/docker/docker/pkg/archive TestIsArchivePathInvalidFile (0.05s)
    archive_test.go:68: Fail to create an archive file for test : dd: failed to open '/Users/ContainerAdministrator/AppData/Local/Temp/TestIsArchivePathInvalidFile1594233703/001/archive': No such file or directory
        .

=== FAIL: github.com/docker/docker/pkg/archive TestIsArchivePathTar (0.05s)
    archive_test.go:84: Fail to create an archive file for test : touch: cannot touch '/Users/ContainerAdministrator/AppData/Local/Temp/TestIsArchivePathTar3066563175/001/archivedata': No such file or directory
        .

```